### ISSUE TITLE
Flutter inited app Crashed iOS 11 real device

### ISSUE THREAD

**Issue by zjhiphop** [flutter/flutter#13773]
## Steps to Reproduce
1. use `flutter create app` to init a app
2. connect device to MAC
3. use `flutter run -d xxx` to run app in real device

The last few lines of `flutter run -verbose`
```
[  +90 ms] (lldb)     command script import "/tmp/7646EC15-30B1-4C78-9521-7A0A960C6696/fruitstrap_4508569e667fe8456bd64e18cf60d6f60016d631.py"
[  +10 ms] (lldb)     command script add -f fruitstrap_4508569e667fe8456bd64e18cf60d6f60016d631.connect_command connect
[        ] (lldb)     command script add -s asynchronous -f fruitstrap_4508569e667fe8456bd64e18cf60d6f60016d631.run_command run
[        ] (lldb)     command script add -s asynchronous -f fruitstrap_4508569e667fe8456bd64e18cf60d6f60016d631.autoexit_command autoexit
[        ] (lldb)     command script add -s asynchronous -f fruitstrap_4508569e667fe8456bd64e18cf60d6f60016d631.safequit_command safequit
[        ] (lldb)     connect
[  +34 ms] (lldb)     run
[ +343 ms] success
[        ] (lldb)     safequit
[ +158 ms] Application launched on the device. Waiting for observatory port.
```
## Logs
```
flutter analyze

Analyzing /Users/jade/Documents/project/learn-flutter/hello...
No issues found!
Ran in 10.1s
```
## Flutter Doctor

[✓] Flutter (on Mac OS X 10.13.2 17C88, locale en-CN, channel alpha)
    • Flutter at /Users/jade/Documents/project/learn-flutter/flutter
    • Framework revision 8f65fec5f5 (2 weeks ago), 2017-12-12 09:50:14 -0800
    • Engine revision edaecdc8b8
    • Tools Dart version 1.25.0-dev.11.0
    • Engine Dart version 2.0.0-edge.d8ae797298c3a6cf8dc9f4558707bd2672224d3e

[✓] Android toolchain - develop for Android devices (Android SDK 25.0.2)
    • Android SDK at /Users/jade/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-25, build-tools 25.0.2
    • ANDROID_HOME = /Users/jade/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_112-release-b06)

[-] iOS toolchain - develop for iOS devices (Xcode 9.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 9.2, Build version 9C40b
    ✗ Verify that all connected devices have been paired with this computer in Xcode.
      If all devices have been paired, libimobiledevice and ideviceinstaller may require updating.
      To update, run:
        brew uninstall --ignore-dependencies libimobiledevice
        brew install --HEAD libimobiledevice
        brew install ideviceinstaller
    • ios-deploy 1.9.2
    • CocoaPods version 1.3.1

[✓] Android Studio (version 2.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Java version OpenJDK Runtime Environment (build 1.8.0_112-release-b06)

[✓] Connected devices
    • Jade • 4508569e667fe8456bd64e18cf60d6f60016d631 • ios • iOS 11.2.1

## Notes
And when i try to run in XCode directly, the console log is: 
```
2017-12-27 18:22:43.232502+0800 Runner[4338:1516407] [Accessibility] ****************** Loading GAX Client Bundle ****************
2017-12-27 18:22:43.786108+0800 Runner[4338:1516407] You've implemented -[<UIApplicationDelegate> application:didReceiveRemoteNotification:fetchCompletionHandler:], but you still need to add "remote-notification" to the list of your supported UIBackgroundModes in your Info.plist.
2017-12-27 18:22:44.403088+0800 Runner[4338:1516613] Observatory listening on http://127.0.0.1:49205/
2017-12-27 18:22:44.470402+0800 Runner[4338:1516407] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '*** -[__NSPlaceholderArray initWithObjects:count:]: attempt to insert nil object from objects[1]'
*** First throw call stack:
(0x185d02364 0x184f48528 0x185c9ae9c 0x185bcdaa0 0x185bd9a7c 0x10084c0c0 0x10084a600 0x18f1fc670 0x18f2653bc 0x18f265358 0x18f451c44 0x18f444ef0 0x18f45cc0c 0x185caa790 0x185caa060 0x185ca7d70 0x185bc7e58 0x187a74f84 0x18f24767c 0x1004f5ce0 0x1856e456c)
libc++abi.dylib: terminating with uncaught exception of type NSException

```

**Comment by xster** [comment]
@cbracken might know

**Comment by solie** [comment]
In contrast with mine, I have successfully run the flutter test app on my iPad pro, while it crashes on iPhone 6 simulator with the following message:
`2018-03-01 13:01:38.176479+0800 Runner[29008:3890785] You've implemented -[<UIApplicationDelegate> application:didReceiveRemoteNotification:fetchCompletionHandler:], but you still need to add "remote-notification" to the list of your supported UIBackgroundModes in your Info.plist.
2018-03-01 13:01:38.370315+0800 Runner[29008:3890903] Observatory listening on http://127.0.0.1:63722/
2018-03-01 13:01:38.974068+0800 Runner[29008:3890785] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '*** -[__NSPlaceholderArray initWithObjects:count:]: attempt to insert nil object from objects[1]'
*** First throw call stack:
(
	0   CoreFoundation                      0x000000010465d12b __exceptionPreprocess + 171
	1   libobjc.A.dylib                     0x0000000103cf1f41 objc_exception_throw + 48
	2   CoreFoundation                      0x000000010469d0cc _CFThrowFormattedException + 194
	3   CoreFoundation                      0x000000010456b23d -[__NSPlaceholderArray initWithObjects:count:] + 237
	4   CoreFoundation                      0x0000000104571e34 +[NSArray arrayWithObjects:count:] + 52
	5   Flutter                             0x0000000101fd4ef0 -[FlutterViewController onLocaleUpdated:] + 153
	6   Flutter                             0x0000000101fd3288 -[FlutterViewController viewDidAppear:] + 116
	7   UIKit                               0x0000000104c8924f -[UIViewController _setViewAppearState:isAnimating:] + 699
	8   UIKit                               0x0000000104c8c0f9 __64-[UIViewController viewDidMoveToWindow:shouldAppearOrDisappear:]_block_invoke + 42
	9   UIKit                               0x0000000104c8a3ed -[UIViewController _executeAfterAppearanceBlock] + 86
	10  UIKit                               0x0000000104ae6e95 _runAfterCACommitDeferredBlocks + 634
	11  UIKit                               0x0000000104ad5bb1 _cleanUpAfterCAFlushAndRunDeferredBlocks + 280
	12  UIKit                               0x0000000104af6278 __34-[UIApplication _firstCommitBlock]_block_invoke_2 + 155
	13  CoreFoundation                      0x000000010460005c __CFRUNLOOP_IS_CALLING_OUT_TO_A_BLOCK__ + 12
	14  CoreFoundation                      0x00000001045e483b __CFRunLoopDoBlocks + 203
	15  CoreFoundation                      0x00000001045e4014 __CFRunLoopRun + 1300
	16  CoreFoundation                      0x00000001045e3889 CFRunLoopRunSpecific + 409
	17  GraphicsServices                    0x000000010a3829c6 GSEventRunModal + 62
	18  UIKit                               0x0000000104adb5d6 UIApplicationMain + 159
	19  Runner                              0x0000000101c9cf2f main + 111
	20  libdyld.dylib                       0x000000010978ed81 start + 1
	21  ???                                 0x0000000000000001 0x0 + 1
)
libc++abi.dylib: terminating with uncaught exception of type NSException
`
## Flutter doctor
`
[✓] Flutter (Channel beta, v0.1.5, on Mac OS X 10.13.4 17E160e, locale en-TW)
[✓] Android toolchain - develop for Android devices (Android SDK 27.0.1)
[✓] iOS toolchain - develop for iOS devices (Xcode 9.2)
[✓] Android Studio (version 3.0)
[✓] VS Code (version 1.20.1)
[✓] Connected devices (1 available)
`

**Comment by cbracken** [comment]
@zjhiphop, @solie thanks for reporting!

It looks like we're missing a `nil` check in FlutterViewController. I'm surprised we're getting a `nil` there though! 

A few questions:
* When did your crash occur? Was it on first startup on on re-foregrounding the app after having switched back to the home screen (or another app)?
*  Is this crash reproducible?
* What locale is your device set to (region/language)?
* Did you happen to switch locales while the app was running?

**Comment by solie** [comment]
Hi @cbracken, in my case the crash occurred every time I want to run it to simulator, right after installation process. It was default to System Language and System Region, and after I change the scheme language/region to english/taiwan in Xcode, it is now running in both simulator and real device.

**Comment by cbracken** [comment]
Thanks for the info. I've landed flutter/engine#4732, which should fix this the next time we roll a dev (or beta) release.
